enterprise/frontend/src/embedding/auth-common/validate-session-token.ts

```diff
@@ -1,35 +1,46 @@
 import * as MetabaseError from "embedding-sdk-bundle/errors";
 
-export function validateSessionToken(session: any) {
-  if (!session || typeof session !== "object") {
+export function validateSessionToken(token: any) {
+  if (!token || typeof token !== "object") {
     throw MetabaseError.INVALID_SESSION_OBJECT({
       expected: "{ jwt: string }",
-      actual: JSON.stringify(session, null, 2),
+      actual: JSON.stringify(token, null, 2),
     });
   }
 
-  if ("status" in session && session.status !== "ok") {
-    if ("message" in session && typeof session.message === "string") {
+  if ("status" in token && token.status !== "ok") {
+    if ("message" in token && typeof token.message === "string") {
       throw MetabaseError.INVALID_SESSION_OBJECT({
         expected: "{ jwt: string }",
-        actual: session.message,
+        actual: token.message,
       });
     }
 
-    if (typeof session.status === "string") {
+    if (typeof token.status === "string") {
       throw MetabaseError.INVALID_SESSION_OBJECT({
         expected: "{ jwt: string }",
-        actual: session.status,
+        actual: token.status,
       });
     }
   }
 
   // We should also receive `iat` and `status` in the response, but we don't actually need them
   // as we don't use them, so we don't throw an error if they are missing
-  if (typeof session.id !== "string" || typeof session.exp !== "number") {
+
+  /**
+   * (EMB-829) Temporarily allow `exp` to be null or undefined while we're deprecating token without it
+   * after we disallow token without expiration, we will re-add this check.
+   */
+  if (typeof token.id !== "string") {
     throw MetabaseError.INVALID_SESSION_SCHEMA({
       expected: "{ id: string, exp: number, iat: number }",
-      actual: JSON.stringify(session, null, 2),
+      actual: JSON.stringify(token, null, 2),
     });
   }
 }
+
+export function shouldRefreshToken(token?: any) {
```

Comment:
logic moved from `enterprise/frontend/src/embedding-sdk-bundle/store/reducer.ts` to make it colocated with the function above. Also this handles the case when there's no token at all (first load).